syntax = "proto3";

option java_multiple_files = true;
option java_package = "pokerEvaluator";
option java_outer_classname = "PokerEvaluatorProto";
option objc_class_prefix = "PKRE";

package pokerEvaluator;

// The service definition.
service Evaluator {
  rpc Evaluate (EvaluationRequest) returns (EvaluationResult) {}
}

// The request message containing command to evaluate
message EvaluationRequest {
  string command = 1;
}

// The response result 
message EvaluationResult {
  message PlayerEvaluationResult{
    enum HandType {
        NoPair = 0;
        OnePair = 1;
        TwoPair = 2;
        Trips = 3;
        Straight = 4;
        Flush = 5;
        FullHouse = 6;
        Quads = 7;
        StraightFlush = 8;
    }
    HandType handType = 1;
    double equityValue = 2;
    double winProbability = 3;
    /* 
    * cardOrder is the ordered player hand resulting from the evaluation from Hi to Lo.
    * The values are not cards.
    * If the result is two pairs, and example is (3 2 K), translate to pair of 3, pair of 2, and K high.
    * For 5 card results, for example Straight, will contain just the highest card. Eg (6), straight to the six.
    */
    string cardOrder = 4; 
  }
  repeated PlayerEvaluationResult results = 1;
}
